### PR TITLE
bitfinex pro parseTrade string math

### PR DESCRIPTION
**Pull request by samgermain**
```
% bitfinex watchTrades BTC/USDT
2023-07-31T22:46:05.311Z
Node.js: v18.15.0
CCXT v4.0.43
bitfinex.watchTrades (BTC/USDT)
    timestamp |                 datetime |   symbol | id | order | type | takerOrMaker | side | price |     amount |          cost | fee | fees
-----------------------------------------------------------------------------------------------------------------------------------------------
1690840814000 | 2023-07-31T22:00:14.000Z | BTC/USDT |    |       |      |              | sell | 29241 | 0.00049838 |   14.57312958 |     |   []
...
1690843504000 | 2023-07-31T22:45:04.000Z | BTC/USDT |    |       |      |              | sell | 29211 |      0.001 |        29.211 |     |   []
30 objects
    timestamp |                 datetime |   symbol |         id | order | type | takerOrMaker | side | price |    amount |         cost | fee | fees
-----------------------------------------------------------------------------------------------------------------------------------------------------
1690843584000 | 2023-07-31T22:46:24.000Z | BTC/USDT | 1400556513 |       |      |              | sell | 29205 |  0.296488 |   8658.93204 |     |   []
...
1690843584000 | 2023-07-31T22:46:24.000Z | BTC/USDT | 1400556510 |       |      |              | sell | 29206 |    0.1054 |    3078.3124 |     |   []
6 objects
    timestamp |                 datetime |   symbol |         id | order | type | takerOrMaker | side | price | amount |      cost | fee | fees
-----------------------------------------------------------------------------------------------------------------------------------------------
1690843584000 | 2023-07-31T22:46:24.000Z | BTC/USDT | 1400556509 |       |      |              | sell | 29206 | 0.1808 | 5280.4448 |     |   []
1 objects
    timestamp |                 datetime |   symbol |         id | order | type | takerOrMaker | side | price | amount | cost | fee | fees
------------------------------------------------------------------------------------------------------------------------------------------
1690843584000 | 2023-07-31T22:46:24.000Z | BTC/USDT | 1400556507 |       |      |              | sell | 29210 |    0.1 | 2921 |     |   []
1 objects
```

### CODE REVIEW

File: js/pro/bitfinex.js [hunk 1]
```diff
@@ -166,16 +166,16 @@ module.exports = class bitfinex extends bitfinexRest {
...
         if (amount !== undefined) {
-            side = (amount > 0) ? 'buy' : 'sell';
-            amount = Math.abs (amount);
+            side = Precise.stringGt (amount, '0') ? 'buy' : 'sell';
+            amount = Precise.stringAbs (amount);
         }
         let cost = undefined;
         if ((price !== undefined) && (amount !== undefined)) {
-            cost = price * amount;
+            cost = Precise.stringMul (amount, price);
```
Comment (frosty00) [inline js/pro/bitfinex.js:178]: if we are using safeTrade we can delete this line

**Comment by frosty00** [comment]
needs to be updated to the typescript version of the lib, if you can do that and also remove the cost calculation will merge

**Comment by samgermain** [comment]
> needs to be updated to the typescript version of the lib, if you can do that and also remove the cost calculation will merge

Finished

**Comment by samgermain** [comment]
reopened in #18742

**Pull request closed without merging**